### PR TITLE
Add Joesecurity onprem integration

### PR DESCRIPTION
**Pull request by rundssoar**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
This is a new pull request, after PR 31576 was somehow "corrupted"
This implements https://xsoar.ideas.aha.io/ideas/FR-I-3623

## Description
Add a Version of the Integration that is compatible with an on-premise instance
The On-Premise Integration is a direct copy of the existing Cloud integration with the following adjustments:
- Removed already deprecated ´export-to-jbxview´ parameter usage
- Remove 'Cloud Only' ´delete-after-days´ parameter 
Details regarding Deprecation / Cloud Only taken from https://github.com/joesecurity/jbxapi/blob/master/jbxapi.py
- The "Detonate File - Joe Sandbox V2" Playbook has been adjusted, to support both Versions of the Integration instead of only the Cloud Version

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by MLainer1** [comment]
Hi @rundssoar, thank you for your contribution.
After talking to our product design team, we prefer that you will add your changes to the original integration, and not create a new integration, as the changes that you added can be inserted to the original integration.
If you need any help with it, feel free to reach out.

**Review by melamedbn — APPROVED** [review]

**Review by MLainer1 — CHANGES_REQUESTED** [review]
Nice work!
Please add a breaking changes .json file to document the deprecated arguments.

When you are ready, let's schedule a quick demo to make sure everything works as expected.

File: Packs/JoeSecurity/ReleaseNotes/1_1_18.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+- *export_to_jbxview* parameter is deprecated and are no longer used
+- Added support for On-Premise Servers
+
+
+#### Playbooks
+
+##### Detonate File - JoeSecurity V2
+
+- Changes not relevant for marketplace
+
```
[the remark this answers is not in the document]
Comment (rundssoar) [reply Packs/JoeSecurity/ReleaseNotes/1_1_18.md]: done

**Comment by rundssoar** [comment]
@MLainer1 I've added the breaking changes json as requested
Unless there is something else missing, i'm ready to do the demo

**Comment by MLainer1** [comment]
Hi @rundssoar, thank you for your time in the demo. 
- Please add a detailed error message when using the premise argument with the cloud configuration, so the user will have a better understanding of the error.
- After checking, a deprecated argument will not appear on the xsoar UI.

Let me know when the PR is ready :)

**Comment by rundssoar** [comment]
@MLainer1 I did implement the more detailed Error message, as discussed.
Also, when looking into that, i came across the fact, that the filename could be given to the API, so i made another small addition, that should have no functional impact: i added a parameter to the joe-submit-sample command, that allows to provide the filename of the sample. 
This is based on the Documentation of the functions: https://github.com/joesecurity/jbxapi/blob/master/docs/api.md
Since i made that after the demonstration, i'm adding a picture depicting the difference between providing and not providing the optional parameter here.
filename_demo
I did add it to the Readme and the Release Notes.
From my point, the PR is ready, unless you have any comments to that last minute additions

File: Packs/JoeSecurity/Integrations/JoeSecurityV2/JoeSecurityV2.yml [hunk 2]
```diff
@@ -27,6 +27,12 @@ configuration:
   - F - Reliability cannot be judged
   required: true
   type: 15
+- name: onprem
+  display: On-Premise
+  type: 8
+  additionalinfo: Only use Features that are available in On-Premise installations
```
Comment (ShirleyDenkberg) [inline Packs/JoeSecurity/Integrations/JoeSecurityV2/JoeSecurityV2.yml:33]:
```suggestion
  additionalinfo: Only use features that are available in On-Premise installations
```

File: Packs/JoeSecurity/ReleaseNotes/1_1_18.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+#### Integrations
+
+##### Joe Security v2
+- Updated the Docker image to: *demisto/vendors-sdk:1.0.0.86440*.
+
+
+- *export_to_jbxview* parameter is deprecated and are no longer used
+- Added support for On-Premise Servers
+- added *file_name* parameter to joe-submit-sample command to allow providing the original filename in submissions
```
Comment (ShirleyDenkberg) [inline Packs/JoeSecurity/ReleaseNotes/1_1_18.md:10]:
```suggestion
- The *export_to_jbxview* parameter is deprecated and is no longer used.
- Added support for On-Premise Servers.
- Added the *file_name* parameter to the ***joe-submit-sample*** command to allow providing the original filename in submissions.
```

**Comment by ShirleyDenkberg** [comment]
@melamedbn @michal-dagan @sapirshuker @MLainer1 Doc review completed.

**Comment by MLainer1** [comment]
Hi @rundssoar, please merge from master, and make sure to update the docker image. When it's done, we can move forward and merge the PR

**Review by MLainer1 — APPROVED** [review]

**Pull request merged**